internal/test/framework/windowsvm.go

```diff
@@ -21,172 +23,157 @@ const (
 type TestWindowsVM interface {
 	// RetrieveDirectories recursively copies the files and directories from the directory in the remote Windows VM
 	// to the given directory on the local host.
 	RetrieveDirectories(string, string) error
-	// Destroy destroys the Windows VM
-	// TODO: Remove this and move it to framework or other higher level object capable of doing deletions.
-	//		jira: https://issues.redhat.com/browse/WINC-243
-	Destroy() error
-	// BuildWMCB returns the value of buildWMCB. It can be used by WSU to decide if it should build WMCB before using it
-	BuildWMCB() bool
-	// SetBuildWMCB sets the value of buildWMCB. Setting buildWMCB to true would indicate WSU will build WMCB instead of
-	// downloading the latest as per the cluster version. False by default
-	SetBuildWMCB(bool)
-	// Compose the Windows VM we have from WNI
-	types.WindowsVM
+	// Compose the Windows VM we have from MachineSets
+	windows.WindowsVM
 }
 
-// newWindowsVM creates and sets up a Windows VM in the cloud and returns the WindowsVM interface that can be used to
-// interact with the VM. If credentials are passed then it is assumed that VM already exists in the cloud and those
-// credentials will be used to interact with the VM. If no error is returned then it is guaranteed that the VM was
-// created and can be interacted with. If skipSetup is true, then configuration steps are skipped.
-func newWindowsVM(imageID, instanceType string, credentials *types.Credentials, skipSetup bool) (TestWindowsVM, error) {
-	w := &testWindowsVM{}
-	var err error
-
-	cloudProvider, err = cloudprovider.CloudProviderFactory(kubeconfig, awsCredentials, "default", artifactDir,
-		imageID, instanceType, sshKey, privateKeyPath)
+// createMachineSet() gets the generated MachineSet configuration from cloudprovider package and creates a MachineSet
+func (f *TestFramework) createMachineSet() error {
+	cloudProvider, err := providers.NewCloudProvider(sshKey)
+	if err != nil {
+		return fmt.Errorf("error instantiating cloud provider %v", err)
+	}
+	machineSet, err := cloudProvider.GenerateMachineSet(true, 1)
+	if err != nil {
+		return fmt.Errorf("error generating Windows MachineSet: %v", err)
+	}
+	log.Print("Creating Machine Sets")
+	_, err = f.machineClient.MachineSets("openshift-machine-api").Create(context.TODO(), machineSet, metav1.CreateOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("error instantiating cloud provider %v", err)
+		return fmt.Errorf("error creating MachineSet %v", err)
 	}
+	f.machineSet = machineSet
+	log.Printf("Created Machine Set %v", machineSet.Name)
+	return nil
+}
 
-	if credentials == nil {
-		windowsVM, err := cloudProvider.CreateWindowsVM()
+// waitForMachines() waits until all the machines required are in Provisioned state. It returns an array of all
+// the machines created. All the machines are created concurrently.
+func (f *TestFramework) waitForMachines(vmCount int, skipVMSetup bool) ([]mapi.Machine, error) {
```

Comment:
nit: can be renamed along the lines of getMachineSets as it does more than just waiting for them to be provisioned

Reply:
changed it to `getWindowsMachines()`